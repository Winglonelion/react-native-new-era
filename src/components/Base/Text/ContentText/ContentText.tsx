import React, { memo } from 'react';
import { ColorValue, StyleSheet, Text, TextProps } from 'react-native';
import { FontFamily } from 'theme/CommonFonts';
import Colors from 'theme/colors';

export interface PropTypes extends TextProps {
  color?: ColorValue;
  size?: number;
  weight?: '100' | '400' | '500' | '600' | 'bold' | 'normal';
  textAlign?: 'left' | 'right' | 'center' | 'auto' | 'justify';
  lineHeight?: number;
}

const ContentText: React.FC<PropTypes> = memo(
  ({
    children,
    style,
    size = 14,
    color = Colors.black,
    weight = 'normal',
    lineHeight = 22,
    textAlign = 'justify',
  }) => {
    return (
      <Text
        style={[
          styles.text,
          {
            fontSize: size,
            color,
            fontWeight: weight,
            lineHeight,
            textAlign: textAlign,
          },
          style,
        ]}>
        {children}
      </Text>
    );
  },
);

const styles = StyleSheet.create({
  text: {
    fontFamily: FontFamily.Helvetica,
  },
});
export default ContentText;
