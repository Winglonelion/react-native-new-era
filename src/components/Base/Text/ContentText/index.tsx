export { default as default } from './ContentText';
