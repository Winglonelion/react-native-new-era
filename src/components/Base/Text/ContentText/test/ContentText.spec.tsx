/* eslint-disable react-native/no-raw-text */
import React from 'react';
import { shallow } from 'enzyme';
import ContentText, { PropTypes } from '../ContentText';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<ContentText {...props}>{'TEST'}</ContentText>);
  return {
    props,
    wrapper,
  };
};

describe('ContentText check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({ color: 'blue', size: 15, weight: '500' });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
