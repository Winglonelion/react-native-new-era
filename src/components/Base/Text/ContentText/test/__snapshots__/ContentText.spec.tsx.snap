// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentText check snapshot should match snapshot with props 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Helvetica Neue",
      },
      Object {
        "color": "blue",
        "fontSize": 15,
        "fontWeight": "500",
        "lineHeight": 22,
        "textAlign": "justify",
      },
      undefined,
    ]
  }
>
  TEST
</Text>
`;

exports[`ContentText check snapshot should match snapshot without props 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Helvetica Neue",
      },
      Object {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "normal",
        "lineHeight": 22,
        "textAlign": "justify",
      },
      undefined,
    ]
  }
>
  TEST
</Text>
`;
